packages/emotion-primitives/src/createEmotion.js

```diff
@@ -0,0 +1,109 @@
+import * as React from 'react'
+import reactPrimitives from 'react-primitives'
+import PropTypes from 'prop-types'
+
+import { getStyles } from './getStyles'
+import { convertToRNStyles } from './convertToRNStyles'
+
+const isValidPrimitive = primitive =>
+  ['Text', 'View', 'Image'].indexOf(primitive) > -1
+
+const getPrimitive = primitive => {
+  if (typeof primitive === 'string' && isValidPrimitive(primitive)) {
+    return reactPrimitives[primitive]
+  } else if (typeof primitive === 'string' && !isValidPrimitive(primitive)) {
+    throw new Error(
+      `Cannot style invalid primitive ${primitive}. Expected primitive to be one of ['Text', 'View', 'Image']`
+    )
+  } else if (typeof primitive === 'function') {
+    return primitive
+  }
+}
+
+function evalStyles(context, Comp, styles, styleOverrides) {
+  // Assign static property so that the styles can be reused (like in withComponent)
+  Comp.styles = convertToRNStyles.call(context, styles)
+
+  return getStyles.call(context, Comp.styles, context.props, styleOverrides)
+}
+
+/**
+ * Creates a function that renders the styles on multiple targets with same code.
+ */
+export function createEmotionPrimitive(splitProps) {
+  /*
+   * Returns styled component
+   */
+  return function emotion(primitive, { displayName } = {}) {
+    return createStyledComponent
+
+    /**
+     * Create emotion styled component
+     */
+    function createStyledComponent() {
+      let styles = []
+
+      styles.push.apply(styles, arguments)
+
+      class Styled extends React.Component {
+        static propTypes = {
+          innerRef: PropTypes.func
+        }
+
+        onRef = innerComponent => {
+          this.innerComponent = innerComponent
+
+          if (this.props.innerRef) {
+            this.props.innerRef(innerComponent)
+          }
+        }
+
+        render() {
+          const { toForward, styleOverrides } = splitProps(
+            primitive,
+            this.props
+          )
+
+          const emotionStyles = evalStyles(this, Styled, styles, styleOverrides)
+
+          return React.createElement(
+            getPrimitive(primitive),
+            {
+              ...toForward,
```

Comment:
`toForward` is a fresh object, it would be good to avoid making additional copy here by just mutating this object with `ref` and `style` props

Reply:
if we go that way then `children` should get attached to that object too instead of passed as 3rd argument

Reply:
Is there any cost ?

Reply:
to the first one - yeah, to the second one (children) there is not, u can disregard my comment, this one doesn't have to be attached to `toForward`. I've checked react's source code and for a single child react will do it by itself so it doesnt really matter if it's done here or on react's side.

Reply:
Ok. I'll tweak this in next commits!